Src/DependencyCollector/Shared/HttpDependenciesParsingTelemetryInitializer.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System;
+    using Channel;
+    using DataContracts;
+    using Extensibility;
+
+    /// <summary>
+    /// Telemetry Initializer that parses http dependencies into well-known types like Azure Storage.
+    /// </summary>
+    public class HttpDependenciesParsingTelemetryInitializer : ITelemetryInitializer
+    {
+        /// <summary>
+        /// If telemetry item is http dependency - converts it to the well-known type of the dependency.
+        /// </summary>
+        /// <param name="telemetry">Telemetry item to convert.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            var httpDependency = telemetry as DependencyTelemetry;
+
+            if (httpDependency != null && httpDependency.Type != null && httpDependency.Type.ToUpperInvariant() == "HTTP")
```

Comment:
Would it be more efficient to do a case insensitive compare than to create a new string that is uppercase?